paasta_tools/chronos_tools.py

```diff
@@ -225,7 +226,17 @@ def get_env(self):
         return [{"name": key, "value": value} for key, value in original_env.iteritems()]
 
     def get_constraints(self):
-        return self.config_dict.get('constraints', [['pool', 'EQUALS', self.get_pool()]])
+        if 'constraints' in self.config_dict:
+            constraints = self.config_dict.get('constraints')
+        else:
+            constraints = self.config_dict.get('extra_constraints', [])
+            constraints.extend(self.get_deploy_constraints())
+            constraints.extend(self.get_pool_constraints())
+        assert isinstance(constraints, list)
+        return [[str(val) for val in constraint] for constraint in constraints]
```

Comment:
> but this isn't the only place where we have a get_something method that actually looks at other attributes of the config

yeah, and I don't like that at all. I've been bitten quite a few times where I've not been able to access the data I actually want because the corresponding ``get_*`` function does some formatting/processing before returning it.

Though I also agree that ``format_chronos_job_dict`` is pretty complex, so I think ``combine_constraints`` is a fine solution.